### PR TITLE
Add location checks to existing test cases

### PR DESCRIPTION
**Pull request by matsl**
## What

Add location checks to existing test cases

## Why

More test are good.

## Note

Refactor the test cases by using a helper function to verify the button attributes. This is local for each test now but could be moved into a general helper maybe.

### CODE REVIEW

File: test/hbut-tests.el [hunk 1]
```diff
@@ -27,6 +27,14 @@ Needed since hyperbole expands all links to absolute paths and
 /tmp can be a symbolic link."
   (should (member tmp '(("/tmp") ("./tmp") ("/private/tmp")))))
 
+(defun hbut-tests--verify-hattr-at-p (actype args loc lbl-key)
+  "Verify the attribute of hbut at point.
+Checks ACTYPE, ARGS, LOC and LBL-KEY."
+  (should (eq (hattr:get (hbut:at-p) 'actype) actype))
```
Comment (rswgnu) [inline test/hbut-tests.el:33]: You should cache the hbut:at-p result and reuse it throughout the attr tests.

File: test/hui-tests.el [hunk 2]
```diff
@@ -27,6 +27,14 @@
 (declare-function hy-test-helpers:consume-input-events "hy-test-helpers")
 
 
+(defun hui-tests--verify-hattr-at-p (actype args loc lbl-key)
+  "Verify the attribute of hbut at point.
+Checks ACTYPE, ARGS, LOC and LBL-KEY."
+  (should (eq (hattr:get (hbut:at-p) 'actype) actype))
```
Comment (rswgnu) [inline test/hui-tests.el:33]: Same as prev commnt.

**Review by rswgnu — CHANGES_REQUESTED** [review]

File: test/hui-tests.el (hunk 2 above)
Comment (matsl) [reply test/hui-tests.el]: Helper merged into one function using keyworded arguments as we talked about.

**Comment by matsl** [comment]
PR is updated. PTAL.

**Review by rswgnu — APPROVED** [review]

**Pull request merged**